BazelProtobufBindings for Parsing ActionGraph

sean7218 opened the pull request:
Add Protobuf Definitions and BazelProtobufBindings for bazel aquery output proto format

- Introduced `analysis_v2.proto` to define Protobuf messages for aquery data, including module information, rule definitions, and provider details.
- Created a README.md file for the Action Parser, outlining setup instructions and usage for generating Swift bindings from Protobuf definitions.
- Implemented unit tests in `BazelProtobufBindingsTests.swift` to validate the parsing of action graph data from Protobuf output, including tests for compiler arguments and action targets.
- Added sample Protobuf data file `actions.pb` for testing purposes.

sean7218 committed:
Add Protobuf Definitions and Action Parser Tests

- Introduced `analysis_v2.proto` to define Protobuf messages for aquery data, including module information, rule definitions, and provider details.
- Created a README.md file for the Action Parser, outlining setup instructions and usage for generating Swift bindings from Protobuf definitions.
- Implemented unit tests in `ActionParserTests.swift` to validate the parsing of action graph data from Protobuf output, including tests for compiler arguments and action targets.
- Added sample Protobuf data file `actions.pb` for testing purposes.

rockbruno commented:
@sean7218 Is it possible to filter this to only include the exact things we're interested in (like the compiler arguments section of swiftcompile/objccompile) or will we always have to ship the entire spec with the tool?

sean7218 commented:
Regarding action parser with proto, the aquery output is filtered with SwiftCompile|ObjcCompile, the parsing will only be done for these actions, I don’t know if there is a way to filter down even more such that only the compiler flags remains

sean7218 committed:
Slim down the binding files generated

  1. remove build.proto and build.pb.swift
  2. remove stardoc_output.proto and stardoc_output.pb.swift
  3. use Target defined in analysis_v2.proto instead of blaze_query.Target

sean7218 committed:
updated NOTICE for swift-protobuf

sean7218 committed:
rename ActionParser to BazelProtobufBindings
